etna/models/nn/rnn.py

```diff
@@ -0,0 +1,223 @@
+        output, (h_n, c_n) = self.layer(encoder_real)
+        forecast = torch.zeros(
+            size=(decoder_real.shape[0], decoder_real.shape[1], 1), dtype=decoder_real.dtype, device=decoder_real.device
+        )  # (batch_size, decoder_length, 1)
+
+        for i in range(decoder_length - 1):
+            output, (h_n, c_n) = self.layer(decoder_real[:, i, None], (h_n, c_n))
+            forecast_point = self.projection(output[:, -1]).flatten()
+            forecast[:, i, 0] = forecast_point
+            decoder_real[:, i + 1, 0] = forecast_point
+
+        output, (h_n, c_n) = self.layer(decoder_real[:, decoder_length - 1, None], (h_n, c_n))
+        forecast_point = self.projection(output[:, -1]).flatten()
+        forecast[:, decoder_length - 1, 0] = forecast_point
+        return forecast
+
+    def configure_optimizers(self):
+        """Optimizer configuration."""
+        optimizer = torch.optim.Adam(self.parameters(), lr=self.lr, **self.optimizer_params)
+        return optimizer
+
+    def step(self, batch: Batch, *args, **kwargs):  # type: ignore
+        """Step for loss computation for training or validation.
+
+        Parameters
+        ----------
+        batch:
+            batch of data
+
+        Returns
+        -------
+        :
+            loss, true_target, prediction_target
+        """
+        encoder_real = batch["encoder_real"].float()  # (batch_size, encoder_length-1, input_size)
+        decoder_real = batch["decoder_real"].float()  # (batch_size, decoder_length, input_size)
+
+        encoder_target = batch["encoder_target"].float()  # (batch_size, encoder_length-1, 1)
+        decoder_target = batch["decoder_target"].float()  # (batch_size, decoder_length, 1)
+
+        decoder_length = decoder_real.shape[1]
+
+        output, (_, _) = self.layer(torch.cat((encoder_real, decoder_real), dim=1))
+
+        target_prediction = output[:, -decoder_length:]
+        target_prediction = self.projection(target_prediction)  # (batch_size, decoder_length, 1)
+
+        target = decoder_target
+
+        loss = self.loss(target_prediction, target)
+        return loss, target, target_prediction
+
+    def make_samples(self, df: pd.DataFrame) -> Iterator[dict]:
+        """Make samples from segment DataFrame."""
+        encoder_length = self.encoder_length
+        decoder_length = self.decoder_length
+
+        def _make(df, start_idx, encoder_length, decoder_length) -> Optional[dict]:
+            sample: Dict[str, Any] = {"target": list(), "encoder_real": list(), "decoder_real": list(), "segment": None}
+            total_length = len(df["target"])
+            total_sample_length = encoder_length + decoder_length
+
+            if total_sample_length + start_idx > total_length:
+                return None
+
+            sample["decoder_real"] = (
+                df.select_dtypes(include=[np.number])
+                .pipe(lambda x: x[["target"] + [i for i in x.columns if i != "target"]])
```

Comment:
No idea what happens here)

Reply:
No, I mean exactly in this line. You get column with "target" and add all the columns without "target", what is the difference between this and just getting all the columns?

Reply:
I need 'target' to be on certain place